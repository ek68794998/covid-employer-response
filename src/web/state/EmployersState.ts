import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}