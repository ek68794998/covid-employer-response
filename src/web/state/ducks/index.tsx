import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});