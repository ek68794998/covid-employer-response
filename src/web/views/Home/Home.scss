@import "../Colors.scss";
@import "../Dimensions.scss";

$division-padding: 16px;

main#home {
	margin: 0 auto;
	width: 100%;

	> div {
		margin: 0 auto;
		padding: 0 $page-padding;
	}

	div.filters {
		box-shadow: 0 0 8px rgba(0, 0, 0, 0.5);
		background: $background;
		padding: $division-padding;
		position: sticky;
		top: 0;
		z-index: 10;
		
		> div {
			margin: 0 auto;
			max-width: ($body-max-width - $page-padding * 2);
			min-width: ($body-min-width - $page-padding * 2);
			width: 50%;
		}
	}

	div.content {
		max-width: $body-max-width;
		padding-top: $division-padding;
	}
}