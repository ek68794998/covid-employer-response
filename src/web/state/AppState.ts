import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}