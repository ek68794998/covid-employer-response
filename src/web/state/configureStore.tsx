import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;