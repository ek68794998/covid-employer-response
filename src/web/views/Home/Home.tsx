import React, { useEffect } from "react";
import { Helmet } from "react-helmet";
import { useDispatch, useSelector } from "react-redux";
import { withRouter } from "react-router-dom";

import { AppState } from "../../state/AppState";
import { getEmployers as dispatchGetEmployers } from "../../state/ducks/employers/actions";
import { getEmployers } from "../../state/ducks/employers/selectors";
import { getStrings } from "../../state/ducks/localization/selectors";

import { EmployerRecord } from "../../../common/EmployerRecord";
import { LocalizedStrings } from "../../../common/LocalizedStrings";

import "./Home.scss";

const Home: React.FC = (): React.ReactElement => {
	const strings: LocalizedStrings = useSelector((state: AppState) => getStrings(state));
	const employers: EmployerRecord[] = useSelector((state: AppState) => getEmployers(state));

	const dispatch: React.Dispatch<any> = useDispatch();

	useEffect(
		() => {
			dispatch(dispatchGetEmployers);
		},
		[]);

	return (
		<main id="home">
			<Helmet>
				<title>{strings.home}</title>
			</Helmet>
			<div className="content">
				{employers && employers.map((e: EmployerRecord) => e.name)}
			</div>
		</main>
	);
};

export default withRouter(Home) as any;
