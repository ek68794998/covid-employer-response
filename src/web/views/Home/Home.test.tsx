import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { mockComponent } from "../../../__tests__/TestUtils";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import Home from "./Home";

jest.mock(
	"../EmployerListSearch/EmployerListSearch",
	() => mockComponent("EmployerListSearch"));

jest.mock(
	"../EmployerList/EmployerList",
	() => mockComponent("EmployerList"));

describe("<Home />", () => {
	test("renders without exploding", () => {
		const store: Store<AppState, AnyAction> = configureStore({
			strings: {
				appTitle: "áƥƥTïƭℓè",
				home: "λô₥è",
			},
		});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<Home />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
