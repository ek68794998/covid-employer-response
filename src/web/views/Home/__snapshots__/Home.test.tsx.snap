// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders without exploding 1`] = `
<main
  id="home"
>
  <div
    className="Home__Filters"
  >
    <div
      className="SearchInput__Container"
    >
      <i
        className="material-icons"
      >
        search
      </i>
      <input
        onInput={[Function]}
        type="search"
      />
    </div>
  </div>
  <div
    className="Home__Content"
  >
    <div
      className="EmployerList__Container--Loading"
    />
  </div>
</main>
`;
